src/main/java/mops/foren/infrastructure/web/ForenController.java

```diff
@@ -137,24 +135,40 @@ public String enterATopic(@PathVariable String forenID,
     /**
      * Mapping to the thread page.
      *
-     * @param forenID  the forum id
-     * @param topicID  the topic id
      * @param threadID the thread id
      * @param model    the model
      * @return The template for the thread
      */
-    @GetMapping("/my-forums/{forenID}/{topicID}/{threadID}")
-    public String displayAThread(@PathVariable String forenID, @PathVariable String topicID,
-                                 @PathVariable String threadID, Model model) {
-
-        ThreadId threadId = new ThreadId(Long.valueOf(threadID));
+    @GetMapping("/thread")
+    public String displayAThread(@RequestParam("threadId") Long threadID, Model model) {
+        ThreadId threadId = new ThreadId(threadID);
         List<Post> posts = this.postService.getPosts(threadId);
         model.addAttribute("threadTitle", this.threadService.getThread(threadId));
         model.addAttribute("posts", posts);
+        model.addAttribute("post", this.postForm);
 
         return "thread2";
     }
 
+    /**
+     * Creating a post and redirecting to the thread page.
+     *
+     * @param token     Keyclock token
+     * @param postForm  a form to create posts.
+     * @param threadIdL The id of the thread the post is in.
+     * @return The template for the thread
+     */
+    @PostMapping("/post/newPost")
+    @RolesAllowed({"ROLE_studentin", "ROLE_orga"})
+    public String newPost(KeycloakAuthenticationToken token, @ModelAttribute PostForm postForm,
+                          @RequestParam("threadId") Long threadIdL) {
+        ThreadId threadId = new ThreadId(threadIdL);
+        User user = this.userService.getUserFromDB(token);
+        Post post = postForm.getPost(user, threadId);
+        this.threadService.addPostInThread(threadId, post);
+        return "redirect:/thread?threadId=" + threadIdL;
```

Comment:
rename "threadIdL" to e.g. threadIdLong or numericalThreadID

Reply:
done